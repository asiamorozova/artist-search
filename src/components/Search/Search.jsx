import React from 'react';
import PropTypes from 'prop-types';

const Search = ({ artist, onChange, onSubmit }) => (
  <form onSubmit={onSubmit}> 
    <label>
      <input type="text" 
        name="search" 
        placeholder="Enter Artist Here" 
        value={artist} 
        onChange={onChange}
      > 
      </input>

    </label>
    <button type="submit">Search</button>
  </form>
);

Search.propTypes = {
  artist: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired

};

export default Search; 

