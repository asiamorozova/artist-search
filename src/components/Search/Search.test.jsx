import React from 'react';
import { shallow } from 'enzyme';
import Search from './Search';

describe('Search component', () => {
  it('utilizes the Search component', () => {
    const wrapper = shallow(<Search />);
    expect(wrapper).toMatchSnapshot();
  });
});
