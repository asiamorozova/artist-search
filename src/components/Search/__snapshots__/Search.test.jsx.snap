// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component utilizes the Search component 1`] = `
<form>
  <label>
    <input
      name="search"
      placeholder="Enter Artist Here"
      type="text"
    />
  </label>
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
